lib/mqtt_msg.c

```diff
@@ -272,6 +272,17 @@ char *mqtt_get_publish_data(uint8_t *buffer, size_t *length)
     return (char *)(buffer + i);
 }
 
+char *mqtt_get_suback_data(uint8_t *buffer, size_t *length)
+{
+    // SUBACK payload length = total length - (fixed header (2 bytes) + variable header (2 bytes))
+    if (*length - 4 > 0) {
```

Comment:
Suggest casting to signed int before subtracting and comparing (might be potentially insecure as `size_t(-4)` is a big number greater than `0`)

Reply:
I should be smarter than that...